snuba/datasets/table_storage.py

```diff
@@ -57,20 +59,9 @@ def get_physical_topic_name(self, slice_id: Optional[int] = None) -> str:
 
         return physical_topic
 
-    @property
+    @cached_property
```

Comment:
I don't think this will work, or at least the result will be very tricky to manage. 
@cached_property caches the response for the duration of the instance, which, for data model object means as long as the pod is not restarted. 

This would make a repartitioning of a topic (adding partitions/removing partitions) more complex as there would be a period of time when the number of partitions in the topic does not reflect what the pod thinks. 

We generally fetch the number of partitions when starting a node that depends on the number of partitions like the scheduler. The repartitioning always triggers a rebalancing. Why not removing this entirely from the `KafkaTopicSpec` and use the api directly upon rebalancing where needed ?